internal/k8s/builder/gateway.go

```diff
@@ -190,7 +190,42 @@ func (b *GatewayDeploymentBuilder) podSpec() corev1.PodSpec {
 		defaultServiceAccount = b.gateway.Name
 	}
 
+	labels := utils.LabelsForGateway(b.gateway)
+
 	return corev1.PodSpec{
+		Affinity: &corev1.Affinity{
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+					{
+						Weight: 3,
+						PodAffinityTerm: corev1.PodAffinityTerm{
+							LabelSelector: &metav1.LabelSelector{
+								MatchLabels: labels,
+							},
+							TopologyKey: `topology.kubernetes.io/region`,
+						},
+					},
```

Comment:
@mikemorris raised a good point in Slack:

> spread across regions may be an unlikely/undesirable deployment due to cross-region data transfer costs and latency - it may be okay for it to be ranked first if deployed like that, but likely not something we would recommend (and iirc we won't even technically support it until agentless work lands due to gossip latency requirements)

Probably best to drop this guy and just go zone -> host.

Reply:
After some discussion with the team, we've decided to ship just the `host` affinity term for now